core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -245,70 +255,72 @@ export class RteHelperModalComponent {
       }
     }
     this.ngbActiveModal.dismiss(true);
+    if (this.rteFieldValues) {
+      this.rteFieldValues.unsubscribe();
+    }
   }
 
   delete(): void {
```

Comment:
What does delete() do in this context and how does it differ from cancel()?

Reply:
Delete is used when you have already added for example a video and press the video button again to change it, then the delete option appears

Reply:
Sorry, not sure I understand -- you're currently in a component-specific modal, right? How can you press the video button from inside the modal?

Reply:
If you add a video, press done on the modal but not save for the entire "card" you are creating, you can press the video button again to change parameters/video input as these are stored when re-opening the component modal, and then the delete() option appears for some components.